### ISSUE TITLE
📝 declare keyword should be in front of public/protected/private

### ISSUE THREAD

**Issue by aovchinn** [biomejs/biome#11500]
### Environment information

just a difference with prettier, related prettier issue https://github.com/prettier/prettier/issues/15868 

**Input:**

```tsx
class A {
  private declare readonly name: string;
}
```

**Output:**

```tsx
class A {
  private declare readonly name: string;
}

```

**Expected behavior:**

```tsx
class A {
  declare private readonly name: string;
}
```

### Configuration

```JSON

```

### Playground link

https://next.biomejs.dev/playground/?indentStyle=space&quoteProperties=preserve&indentWidth=4&arrowParentheses=as-needed&tab=formatter&pane=Diagnostics#code=YwBsAGEAcwBzACAAQQAgAHsACgAgACAAIABwAHIAaQB2AGEAdABlACAAZABlAGMAbABhAHIAZQAgAHIAZQBhAGQAbwBuAGwAeQAgAG4AYQBtAGUAOgAgAHMAdAByAGkAbgBnADsACgB9AA%3D%3D

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by HarperZ9** [comment]
I'd like to take this if it's open. The cause is the modifier precedence enum in `crates/biome_js_syntax/src/modifier_ext.rs`: `Declare` sits after the accessibility bits (`Private`/`Protected`/`Public`), and the formatter sorts by that order in `sort_modifiers_by_precedence`, so it rewrites `declare private` into `private declare`. Moving `Declare` ahead of accessibility fixes it.

Reordering that bitflag looks safe. The only behavioral change is property signatures, the one modifier list where `declare` appears. Other consumers just test membership with `.contains(...)`, and `useNamingConvention` maps to its own `RestrictedModifier` by variant rather than raw bits, so nothing keys off the numeric order. There's also already a spec that locks in the wrong output today (`ts/declaration/class.ts` turns `declare private readonly` into `private declare readonly`), so it doubles as the regression test once the snapshot is regenerated.

Can you assign it to me?

**Comment by dyc3** [comment]
just make a pr

**Comment by webdevpraveen** [comment]
Hi there! I'd like to take a crack at this. 

I understand the `declare` token needs to be emitted before accessibility modifiers (`public`/`private`/`protected`) in the `biome_js_formatter` logic. I'll take a look at the sorting/formatting logic for TypeScript class members and raise a PR. 

Could you assign this to me?
